### PR TITLE
reduce the nesting of a couple of match statements

### PR DESCRIPTION
**Pull request by anastygnome**
#3062 rebased

### CODE REVIEW

File: src/uu/hashsum/src/hashsum.rs [hunk 1]
```diff
@@ -21,8 +21,8 @@ use self::digest::Digest;
 use self::digest::DigestWriter;
 
 use clap::{App, AppSettings, Arg, ArgMatches};
-use hex::encode;
-use md5::Md5;
+use hex::ToHex;
+use md5::Context as Md5;
```
Comment (tertsdiepraam) [inline src/uu/hashsum/src/hashsum.rs:25]: I think this has recently changed and should be kept how it is now.

File: src/uu/hashsum/src/hashsum.rs [hunk 2]
```diff
@@ -652,6 +644,6 @@ fn digest_reader<T: Read>(
         let mut bytes = Vec::new();
         bytes.resize((output_bits + 7) / 8, 0);
         digest.result(&mut bytes);
-        Ok(encode(bytes))
+        Ok(bytes.to_hex())
```
Comment (tertsdiepraam) [inline src/uu/hashsum/src/hashsum.rs:647]: Same here

**Review by tertsdiepraam — CHANGES_REQUESTED** [review]
Thanks for rebasing it! In the future you can just (force) push the rebased branch to the original PR. I think the rebasing of hashsum introduced some mistakes though, could you take a look at that?

File: src/uu/hashsum/src/hashsum.rs (hunk 2 above)
Comment (anastygnome) [reply src/uu/hashsum/src/hashsum.rs:647]: will co

**Comment by anastygnome** [comment]
@tertsdiepraam Do you want me to squash that ?

**Comment by tertsdiepraam** [comment]
It's fine to just add a commit to fix it

**Comment by anastygnome** [comment]
@tertsdiepraam it's all good

File: src/uu/hashsum/src/hashsum.rs [hunk 3]
```diff
@@ -442,7 +435,6 @@ pub fn uu_app_custom<'a>() -> App<'a> {
             "work with SHAKE256 using BITS for the output size",
         ),
         ("b2sum", "work with BLAKE2"),
-        ("b3sum", "work with BLAKE3"),
```
Comment (tertsdiepraam) [inline src/uu/hashsum/src/hashsum.rs:445]: and here

File: src/uu/hashsum/src/hashsum.rs [hunk 4]
```diff
@@ -70,7 +70,6 @@ fn is_custom_binary(program: &str) -> bool {
             | "shake128sum"
             | "shake256sum"
             | "b2sum"
-            | "b3sum"
```
Comment (tertsdiepraam) [inline src/uu/hashsum/src/hashsum.rs:73]: This should not be removed

File: src/uu/hashsum/src/hashsum.rs [hunk 5]
```diff
@@ -94,40 +93,36 @@ fn detect_algo(
             Box::new(blake2b_simd::State::new()) as Box<dyn Digest>,
             512,
         ),
-        "b3sum" => (
-            "BLAKE3",
-            Box::new(blake3::Hasher::new()) as Box<dyn Digest>,
-            256,
-        ),
```
Comment (tertsdiepraam) [inline src/uu/hashsum/src/hashsum.rs:101]: Same here

**Review by tertsdiepraam — CHANGES_REQUESTED** [review]
I found a few more issues with hashsum. Sorry that this took long to merge and caused conflicts :)

**Comment by anastygnome** [comment]
@tertsdiepraam it should now be all set

**Comment by sylvestre** [comment]
Looks like it doesn't work:
```

---- test_hashsum::b3sum::test_check stdout ----
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils hashsum --b3sum --bits=256 --check b3sum.checkfile
thread 'test_hashsum::b3sum::test_check' panicked at 'Command was expected to succeed.
stdout = 
 stderr = hashsum: You must specify hash algorithm!
', tests/common/util.rs:174:9

```

**Comment by sylvestre** [comment]
ping?

**Comment by sylvestre** [comment]
ping?

**Comment by sylvestre** [comment]
Closed
please reopen when ready

**Comment by anastygnome** [comment]
won't be able to fix this, this is beyond my ability

**Pull request closed without merging**